src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js

```diff
@@ -97,44 +104,123 @@ class EditBaseline extends Component {
         return expandIcon;
     }
 
+    onSelect = (isSelected, event) => {
+        const { editBaselineTableData, selectFact } = this.props;
+        let facts = [];
+
+        if (event.target.name === 'select-all') {
+            this.setState({ selectAll: isSelected });
+            editBaselineTableData.forEach(function(fact) {
+                facts.push(fact[0]);
+                if (editBaselineHelpers.isCategory(fact)) {
+                    editBaselineHelpers.baselineSubFacts(fact).forEach(function(subFact) {
+                        facts.push(subFact[0]);
+                    });
+                }
+            });
+        } else {
+            let factData;
+            let id = Number(event.target.name);
+
+            editBaselineTableData.forEach(function(fact) {
+                if (fact[0] === id) {
+                    factData = fact;
+                } else if (editBaselineHelpers.isCategory(fact)) {
+                    editBaselineHelpers.baselineSubFacts(fact).forEach(function(subFact) {
+                        if (subFact[0] === id) {
+                            factData = subFact;
+                        }
+                    });
+                }
+            });
+
+            facts = [ factData[0] ];
+
+            if (editBaselineHelpers.isCategory(factData)) {
+                editBaselineHelpers.baselineSubFacts(factData).forEach(function(subFact) {
+                    facts.push(subFact[0]);
+                });
+            }
+        }
+
+        selectFact(facts, isSelected);
+    }
+
+    renderCheckbox = (fact) => {
+        const { selectAll } = this.props;
+        let id;
+
+        if (typeof(fact[2]) === 'string') {
+            id = 'fact-' + fact[0];
+        } else if (Array.isArray(fact[2]) && fact[2].length > -1) {
```

Comment:
This can use the new helper functions

Reply:
It can't because the helper function looks for categories that have subFacts. This is checking for a category that has at least an empty array. I could create two different helper functions, isCategory and isCategoryWithFacts.